### ISSUE TITLE
[Feature] Better Docgen for Fields (Objects + Enums)

### ISSUE THREAD

**Issue by awr1** [nim-lang/Nim#12353]
Right now the state of docgen for fields in objects or enums is pretty simple and can roughly be amounted to "copy paste" into a code block (plus a `nimpretty` pass, I'm assuming). Doc comments to the side of a field get "rendered" but only to the effect that they are put in the code block, they aren't allowed to be actually rendered with Markdown.

Doxygen, by contrast, fully renders a list of the fields and their comments: [example here](http://portaudio.com/docs/v19-doxydocs/structPaVersionInfo.html). I like how Doxygen renders the fields as a list (and not as a code block) but I feel like the field documentation could have been done in-line with the list.

In addition, (feel free to prove me wrong, I haven't tried messing with this in a while) I don't think there is a proper way to insert documentation nodes in macros to have them show up in the docs (either that, or it was that doc comments for fields don't show up when try you acquire the type impl).

**Comment by Araq** [comment]
I agree we should improve the docgen, however, keep in mind that ``case objects`` and ``when`` inside objects make this significantly harder than just "render it as a table".

**Comment by FedericoCeratto** [comment]
Related to #4517

**Comment by elijahr** [comment]
I think a few tweaks to the existing output would go a long way.

Given the following code:

```nim
type
  Foo* = object
    ## The foo type
    bar*: int ## set by `setBar <#setBar,Foo>`_
    baz*: int
      ## Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
      ## tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim 
      ## veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
      ## commodo consequat. Duis aute irure dolor in reprehenderit in voluptate
      ## velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint
      ## occaecat cupidatat non proident, sunt in culpa qui officia deserunt
      ## mollit anim id est laborum.  
```

The doc output ends up as:

```
Foo = object
  bar*: int                    ## set by `setBar <#setBar,Foo>`_
  baz*: int ## Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
          ## tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim 
          ## veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
          ## commodo consequat. Duis aute irure dolor in reprehenderit in voluptate
          ## velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint
          ## occaecat cupidatat non proident, sunt in culpa qui officia deserunt
          ## mollit anim id est laborum.  
```

As you can see, the space between `bar` and its comment is inexplicably long, and the newline between the `baz` and its comment is lost. The link in `bar`'s comment is also not rendered.

Output like the following would be easier to read:

```
Foo = object
  bar*: int ## set by setBar
  baz*: int
    ## Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
    ## tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim 
    ## veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
    ## commodo consequat. Duis aute irure dolor in reprehenderit in voluptate
    ## velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint
    ## occaecat cupidatat non proident, sunt in culpa qui officia deserunt
    ## mollit anim id est laborum.  
```

The spacing between fields and comments is retained, and the link to setBar would ideally be rendered.

**Comment by awr1** [comment]
I think at the very least the way the docgen renders doc comments with unnecessary spacing needs to be fixed. I don't use nimsuggest but also making sure enum/object documentation shows up correctly when using nimsuggest is also something to consider.

**Comment by awr1** [comment]
For a better solution, I think maybe what we could do is render doc comments after the line in which they appear and then indent appropiately, something like:
![image](https://user-images.githubusercontent.com/41453959/103490086-01e89e80-4ddf-11eb-9b3c-24399ff764e6.png)
